export-ingester/export_ingester/sftp.py

```diff
@@ -7,9 +7,18 @@
 from .models import ExportList, ExportModel
 
 
-class SFTPClient:
-    def __init__(self, client: asyncssh.SFTPClient):
-        self.client = client
+class SSHClient:
+    def __init__(self, conn: asyncssh.SSHClientConnection):
+        self.conn = conn
+
+    async def remote_sbatch(self, sbatch_command: str, remote: str) -> str:
```

Comment:
- It's a bit unclear what remote is in this context. Isn't this part of the sbatch command?
- What is the intended return of this function? A job id? I think you should be doing some kind of validating to make sure it's correct.
- There isn't any checking as to wether this is a valid sbatch command. What command(s) are you trying to run here?

Reply:
- I guess it's remote because the sbatch command is called on snellius and not locally.
- It could return a job id, yes so at least we know that some job has been launched (and if not that probably something is wrong with snellius)
- How would you recommend doing that? The command is something like sbatch /gpfs/home4/mkerrwinter/orchestrator/orchestrator/bash_scripts/downloader.sh /scratch-shared/amftrack2024/daily (to be defined by user in the .env)

Reply:
Maybe something like this:

```python
from asyncssh import SSHClientConnection
from textwrap import dedent

class SbatchClient:
    DOWNLOAD_CMD = dedent(
        """\
        sbatch --parseable \\
            $HOME/orchestrator/scripts/downloader.sh {file_path}
            /
        """
    )

    def __init__(self, client: SSHClientConnection):
        self.client = client

    async def run_download(
        self,
        src_file: Path,
    ) -> int:
        result = await self.client.run(
            self.DOWNLOAD_CMD.format(file_path=src_file),
            check=True,
        )
        return int(result.stdout)


```